### PR TITLE
[2.4] meson: Simplify options semantics and standardize feature macros

### PR DESCRIPTION
**Pull request by dgsga**
This changeset:

- Simplifies the syntax of the build options to make them more user-friendly
- Standardises the syntax of the feature macros so they all share the same logic
- Fixes the logic of the largefile support macro
- Disables gssapi support if the Kerberos V UAM is not required
- Uses the recommended syntax for compiling and installing (using meson compile and meson install rather than their ninja equivalents)
- Converts the install guide to markdown format for easy reading
- Backports the Netatalk 3 option syntax for initscripts

### CODE REVIEW

**Review by rdmark — APPROVED** [review]

**Comment by rdmark** [comment]
@dgsga Requesting a few documentation fixes; looks good otherwise.

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+options.
+
+## External software dependencies
+
+Required:
+
+  - libevent 2
+  - libgcrypt version 1.2.3 or later (for DHX2 UAM support)
+  - Berkeley DB version 4.6.0 or later
+  - perl
```
[the remark this answers is not in the document]
Comment (dgsga) [reply INSTALL.md]: Fixed

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+Required for Spotlight support:
+  - talloc
+  - tracker version 0.12 or later
+  - bison
+  - flex
```
[the remark this answers is not in the document]
Comment (dgsga) [reply INSTALL.md]: Fixed

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+Optional:
+
+  - avahi or mDNSresponder  (for Zeroconf support)
+  - cracklib                (for cracklib support)
+  - Docbook XSL stylesheets (for manpages & manual documentation)
+  - GLib 2 and dbus-glib    (for afpstats support)
```
[the remark this answers is not in the document]
Comment (dgsga) [reply INSTALL.md]: Fixed

File: INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  - libpam                  (for PAM support)
+  - libressl or OpenSSL@1.1 (for Randnum and DHX UAM support.
+                             Built-in SSL supportis also available)
+  - libtirpc / libquota     (for Quota support)
+  - perl                    (for various scripts)
+  - tcpwrap                 (for TCP wrapper support)
+  - xsltproc                (for manpages and mananual documentation)
+
+More details about dependencies can be found in the documentation at
+https://netatalk.io/stable/htmldocs/installation
```
[the remark this answers is not in the document]
Comment (dgsga) [reply INSTALL.md]: Fixed

File: INSTALL.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+# Compiling with autotools (deprecated)
+
+The rest of this document contains the GNU autotools install
+insructions....
```
[the remark this answers is not in the document]
Comment (dgsga) [reply INSTALL.md]: Fixed

File: INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+Please see `meson_options.txt` for full details of all Netatalk-specific options,
+and https://mesonbuild.com/Builtin-options.html for details of generic Meson
+options.
+
+## External software dependencies
+
+Required:
+
+  - libevent 2
```
[the remark this answers is not in the document]
Comment (dgsga) [reply INSTALL.md]: Fixed

File: INSTALL.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+  - Berkeley DB version 4.6.0 or later
+  - perl
+
+Required for Spotlight support:
+  - talloc
+  - tracker version 0.12 or later
+  - bison
+  - flex
+
+Optional:
```
[the remark this answers is not in the document]
Comment (dgsga) [reply INSTALL.md:85]: Fixed

**Comment by dgsga** [comment]
> @dgsga Requesting a few documentation fixes; looks good otherwise.

@rdmark, All done, thanks for noticing these.

**Comment by rdmark** [comment]
@dgsga You're welcome. Let's merge this then.

**Comment by dgsga** [comment]
@rdmark, Please can you squash the last 3 commits and then do a rebase merge so the commits are more traceable?

**Comment by rdmark** [comment]
Done!

**Pull request merged**